GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,421 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using GoRogue.Random;
 using JetBrains.Annotations;
 using SadRogue.Primitives;
-using Troschuetz.Random;
+using System.Diagnostics.CodeAnalysis;
 
 namespace GoRogue.MapGeneration
 {
     /// <summary>
     /// A region of the map with four sides of arbitrary shape and size
     /// </summary>
     [PublicAPI]
-    public partial class Region : IMatchable<Region>
+    public class Region : IReadOnlyArea
     {
-        #region Generation
+        private Point _northWestCorner;
+        private Point _northEastCorner;
+        private Point _southEastCorner;
+        private Point _southWestCorner;
+        private MultiArea _points;
+        private MultiArea _outerPoints;
+        private Area _innerPoints;
+        private Area _southBoundary;
+        private Area _northBoundary;
+        private Area _eastBoundary;
+        private Area _westBoundary;
 
         /// <summary>
         /// A region of the map with four corners of arbitrary shape and size
         /// </summary>
-        /// <param name="name">the name of this region</param>
         /// <param name="northWest">the North-West corner of this region</param>
         /// <param name="northEast">the North-East corner of this region</param>
         /// <param name="southEast">the South-East corner of this region</param>
         /// <param name="southWest">the South-West corner of this region</param>
-        public Region(string name, Point northWest, Point northEast, Point southEast, Point southWest)
+        public Region(Point northWest, Point northEast, Point southEast, Point southWest)
         {
-            Name = name;
-            SouthEastCorner = southEast;
-            NorthEastCorner = northEast;
-            NorthWestCorner = northWest;
-            SouthWestCorner = southWest;
-
+            _southEastCorner = southEast;
+            _northEastCorner = northEast;
+            _northWestCorner = northWest;
+            _southWestCorner = southWest;
             _westBoundary = new Area(Lines.Get(NorthWestCorner, SouthWestCorner));
             _southBoundary = new Area(Lines.Get(SouthWestCorner, SouthEastCorner));
             _eastBoundary = new Area(Lines.Get(SouthEastCorner, NorthEastCorner));
             _northBoundary = new Area(Lines.Get(NorthEastCorner, NorthWestCorner));
```

Comment:
`Lines.Get` offers several different algorithms for creating lines.  Is there ever a use case where the user might want to specify which ones to use?

Reply:
Yes there is, adding that possibility in now